micromamba/src/shell.cpp

```diff
@@ -8,127 +8,337 @@
 #include "mamba/api/shell.hpp"
 #include "mamba/core/fsutil.hpp"
 #include "mamba/core/run.hpp"
+#include "mamba/core/shell_init.hpp"
 
 #include "common_options.hpp"
 #include "umamba.hpp"
 
 
 using namespace mamba;  // NOLINT(build/namespaces)
 
-void
-init_shell_parser(CLI::App* subcom)
+namespace
 {
-    init_general_options(subcom);
+    void init_shell_parser(CLI::App* subcmd)
+    {
+        init_general_options(subcmd);
 
-    auto& config = Configuration::instance();
+        auto& config = Configuration::instance();
 
-    auto& shell_type = config.insert(
-        Configurable("shell_type", std::string("")).group("cli").description("A shell type")
-    );
-    subcom
-        ->add_option("-s,--shell", shell_type.get_cli_config<std::string>(), shell_type.description())
-        ->check(CLI::IsMember(std::set<std::string>(
-            { "bash", "posix", "powershell", "cmd.exe", "xonsh", "zsh", "fish", "tcsh", "dash" }
-        )));
-
-    auto& stack = config.insert(Configurable("shell_stack", false)
-                                    .group("cli")
-                                    .description("Stack the environment being activated")
-                                    .long_description(unindent(R"(
+        auto& shell_type = config.insert(
+            Configurable("shell_type", std::string("")).group("cli").description("A shell type"),
+            true
+        );
+        subcmd
+            ->add_option("-s,--shell", shell_type.get_cli_config<std::string>(), shell_type.description())
+            ->check(CLI::IsMember(std::set<std::string>(
+                { "bash", "posix", "powershell", "cmd.exe", "xonsh", "zsh", "fish", "tcsh", "dash" }
+            )));
+
+        auto& prefix = config.insert(
+            Configurable("shell_prefix", std::string(""))
+                .group("cli")
+                .description("The root prefix to configure (for init and hook), and the prefix "
+                             "to activate for activate, either by name or by path"),
+            true
+        );
+        subcmd->add_option(
+            "prefix,-p,--prefix,-n,--name",
```

Comment:
Oh so at the end, we keep handling `--name` and `--prefix` with the same option. As long as it works and isn't confusing...

Reply:
The goal is to split them, but this PR was already enough work to introduce the necessary structure to be able to split them. This will be done in a follow-up.